src/backend/metal/RenderPipelineMTL.mm

```diff
@@ -92,24 +92,16 @@ MTLIndexType MTLIndexFormat(nxt::IndexFormat format) {
             }
         }
 
-        RenderPass* renderPass = ToBackend(GetRenderPass());
-        auto& subpassInfo = renderPass->GetSubpassInfo(GetSubPass());
-
-        if (subpassInfo.depthStencilAttachmentSet) {
-            const auto& attachmentInfo =
-                renderPass->GetAttachmentInfo(subpassInfo.depthStencilAttachment);
-            descriptor.depthAttachmentPixelFormat = MetalPixelFormat(attachmentInfo.format);
-            descriptor.stencilAttachmentPixelFormat = MetalPixelFormat(attachmentInfo.format);
+        if (HasDepthStencilAttachment()) {
+            nxt::TextureFormat depthStencilFormat = GetDepthStencilFormat();
+            descriptor.depthAttachmentPixelFormat = MetalPixelFormat(depthStencilFormat);
+            descriptor.stencilAttachmentPixelFormat = MetalPixelFormat(depthStencilFormat);
```

Comment:
I know this was already here, but it looks like it will be wrong for depth-only or stencil-only formats (which I think we don't currently have). Could you add a TODO? (you can assign it to me)

Reply:
Done